Code/GraphMol/ChemTransforms/MolFragmenter.cpp

```diff
@@ -1035,8 +1062,75 @@ std::unique_ptr<ROMol> molzip(const ROMol &a, const ROMol &b,
   return newmol;
 }
 
+RDKIT_CHEMTRANSFORMS_EXPORT std::unique_ptr<ROMol> molzip(
+    const ROMol &a, const ROMol &b, const MolzipParams &params) {
+  std::optional<std::map<int, int>> opt(std::nullopt);
+  return molzip(a, b, params, opt);
+}
+
 std::unique_ptr<ROMol> molzip(const ROMol &a, const MolzipParams &params) {
   const static ROMol b;
   return molzip(a, b, params);
 }
+
+std::unique_ptr<ROMol> molzip(std::vector<ROMOL_SPTR> &decomposition,
+                              const MolzipParams &params) {
+  if (params.generateCoordinates) {
+    int index = 0;
+    for (const auto &mol : decomposition) {
+      for (const auto atom : mol->atoms()) {
+        atom->setProp(indexPropName, index++);
+      }
+    }
+  }
+
+  const auto combinedMol = std::accumulate(
```

Comment:
Not for this PR, but this could be a lot more efficient if we had an overload of `combineMols()`:
`void combineMols(RWMol &mol1, const ROMol &mol2)`